### PR TITLE
s3 list

### PR DESCRIPTION
**Pull request by fparisotto**

### CODE REVIEW

File: src/main/scala/ignition/core/jobs/utils/SparkContextUtils.scala [hunk 1]
```diff
@@ -26,6 +28,11 @@ object SparkContextUtils {
     override def getPartition(key: Any): Int = index(key)
   }
 
+  implicit class S3ObjectSummaryExtensions(s3Object: S3ObjectSummary) {
+    def toHadoopFile: HadoopFile =
+      HadoopFile(s"s3n://${s3Object.getBucketName}/${s3Object.getKey}", isDir = false, s3Object.getSize)
+  }
```
Comment (sisso) [inline src/main/scala/ignition/core/jobs/utils/SparkContextUtils.scala]: I can not say that I like explicit classes. Specially for a single usage. But is me :P

File: src/main/scala/ignition/core/utils/S3Utils.scala [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            (implicit s3: AmazonS3Client): List[S3ObjectSummary] = {
+    def inner(acc: mutable.ArrayBuffer[S3ObjectSummary], listing: ObjectListing): List[S3ObjectSummary] = {
+      acc ++= listing.getObjectSummaries.toList.filter(predicate)
+      if (listing.isTruncated)
+        inner(acc, s3.listNextBatchOfObjects(listing))
+      else
+        acc.toList
+    }
+
+    inner(new mutable.ArrayBuffer[S3ObjectSummary], s3.listObjects(bucket, prefix))
```
Comment (sisso) [inline src/main/scala/ignition/core/utils/S3Utils.scala]: I am not against it, but others should be :P

File: src/main/scala/ignition/core/utils/S3Utils.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  def s3ListAndFilterFiles(bucket: String,
+                           prefix: String,
+                           start: Option[DateTime] = None,
+                           end: Option[DateTime] = None,
+                           endsWith: Option[String] = None,
+                           exclusionPattern: Option[String] = Option("_$folder$"),
+                           predicate: S3ObjectSummary => Boolean = _ => true)
+                          (implicit s3: AmazonS3Client, pathDateExtractor: PathDateExtractor): List[S3ObjectSummary] = {
+
+    def excludePatternValidation(s3Object: S3ObjectSummary, exclusionPatternOption: Option[String]): Option[S3ObjectSummary] =
```
Comment (sisso) [inline src/main/scala/ignition/core/utils/S3Utils.scala]: I don't know if it helps. But I think that most of this code could be simplified by filters (S3ObjectSummary) => Boolean

**Comment by sisso** [comment]
LGTM, only shit comments. 

As always, I don't know little about this.

File: src/main/scala/ignition/core/jobs/utils/SparkContextUtils.scala (hunk 1 above)
Comment (fparisotto) [inline src/main/scala/ignition/core/jobs/utils/SparkContextUtils.scala]: You're right

**Comment by douglaz** [comment]
LGTM.

File: src/main/scala/ignition/core/utils/S3Utils.scala [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+  def s3ListAndFilterFiles(bucket: String,
+                           prefix: String,
+                           start: Option[DateTime] = None,
+                           end: Option[DateTime] = None,
+                           endsWith: Option[String] = None,
+                           exclusionPattern: Option[String] = Option("_$folder$"),
+                           predicate: S3ObjectSummary => Boolean = _ => true)
+                          (implicit s3: AmazonS3Client, pathDateExtractor: PathDateExtractor): List[S3ObjectSummary] = {
```
Comment (douglaz) [inline src/main/scala/ignition/core/utils/S3Utils.scala]: this method should work out of box without a explicit s3 client.

**Comment by douglaz** [comment]
LGTM.

**Pull request merged**